ocp_resources/pipelineruns.py

```diff
@@ -0,0 +1,74 @@
+# API reference: https://tekton.dev/docs/pipelines/pipelineruns/
+
+from ocp_resources.resource import NamespacedResource
+
+
+class PipelineRun(NamespacedResource):
+    api_group = NamespacedResource.ApiGroup.TEKTON_DEV
+
+    def __init__(
+        self,
+        name,
+        namespace,
+        win_image_download_url,
+        pipelineref,
+        params=None,
+        service_account_name=None,
+        client=None,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the pipelinerun
+            namespace (str): Namespace of the pipelinerun
+            win_image_download_url (str): Mandatory: Source to get windows image.
+            pipelineref (str): Mandatory: Base pipeline to run pipelineruns
+            client: (DynamicClient): DynamicClient to use.
+            params (dict): Optional params to add during triggering a run.
+            params can be set/changed based on pipelineref.
+            service_account_name (str): Optional to provide service account
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            **kwargs,
+        )
+        self.pipelineref = pipelineref
+        self.win_image_download_url = win_image_download_url
+        self.params = params
+        self.service_account_name = service_account_name
+
+    @property
+    def _subresource_api_url(self):
+        return (
+            f"{self.client.configuration.host}/"
+            f"apis/tekton.dev/{self.api.api_version}/"
+            f"namespaces/{self.namespace}/pipelineruns"
+        )
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res["spec"] = {}
+            if self.pipelineref:
+                self.res["spec"]["pipelineref"] = {
+                    "name": self.pipelineref,
+                }
+
+            if self.win_image_download_url:
+                self.res["spec"]["params"] = [
+                    {
+                        "name": "winImageDownloadURL",
+                        "value": self.win_image_download_url,
+                    }
+                ]
```

Comment:
see comment above - "params" should get this list of dicts and not address private use cases; see https://tekton.dev/docs/pipelines/pipelineruns/#specifying-parameters